packages/data/src/api/nearPrice/nearPrice.ts

```diff
@@ -8,25 +8,20 @@ import {
 } from './nearPrice.types';
 
 
-export const fetchPriceFromCoinGecko = async (): Promise<NearPriceData> => {
-  const req = await fetch(COIN_GECKO_API);
-  const data = await req.json() as CoinGeckoNearPriceData;
-  return { price: data.near.usd };
-};
-
-export const fetchPriceFromBinance = async (): Promise<NearPriceData> => {
-  const req = await fetch(BINANCE_API);
-  const data = await req.json() as NearPriceData;
-  return data;
-};
+async function fetchPrice<T>(url: string, tokenPrice: (data: T) => NearPriceData): Promise<NearPriceData> {
+  const req = await fetch(url);
+  const data = await req.json() as T;
+  return tokenPrice(data);
+}
 
 export const nearPrice = async (): Promise<ParsedDataReturn<string>> => {
   try {
-    const res = await Promise.any([
-      fetchPriceFromCoinGecko(),
-      fetchPriceFromBinance(),
-    ]);
-
+    let res: NearPriceData;
+    try {
+      res = await fetchPrice<CoinGeckoNearPriceData>(COIN_GECKO_API, data => ({ price: data.near.usd }));
+    } catch (err) {
+      res =  await fetchPrice<NearPriceData>(BINANCE_API, data => data);
+    }
```

Comment:
You are prioritizing CoinGecko by calling it first and swallowing their errors. 

These two APIS are different. Just keep the Promise.any, catch rate limit errors and handle them differently (like return null). 

Have you considered adding another API? 

https://coinmarketcap.com/api/pricing/

https://coinpaprika.com/api/

Reply:
```
Have you considered adding another API?

https://coinmarketcap.com/api/pricing/

https://coinpaprika.com/api/
```

this is something I needed to do later, user can pass it api and we fetch for him.

Reply:
<img width="293" alt="Screenshot 2024-07-15 at 11 00 11" src="https://github.com/user-attachments/assets/c97af0e7-d038-4fd1-9c94-52a2cb562491">